### PR TITLE
[mail] improve attachments and fix feature

### LINKED ISSUE #6423: [mail] Unable to send email with an attachment
I'm unable to send an email with an attachment using the OH2 mail binding. I'm on build 1750.

When executing the code below, I get two emails. The first email correctly contains the supplied subject and content text. The second email contains the supplied subject text, but no content and no attachment.

I’ve tried this with two different mail servers with the same result.

```
    val String imageUrl = "http://my.openhab.host:8080/static/path/to/image.jpg"
    //val String imageUrl = "file:///path/to/another/image.jpg"
    //val String imageUrl = "file:///path/to/a/text/file"

    //val mailActions = getActions("mail","mail:smtp:gmail")
    val mailActions = getActions("mail","mail:smtp:anothermailserver")

    mailActions.sendMail("me@somewhere.com", "Email without attachment", "Email content without attachment")

    mailActions.sendMail("me@somewhere.com", "Email with attachment", "Email content with attachment", imageUrl)
```

When viewing the raw source of the second received email, this is what I see.
```
[snip mail headers]
Subject: Email with attachment
MIME-Version: 1.0
Content-Type: multipart/mixed; 
	boundary="----=_Part_7_1653710751.1573740395359"
```

I've installed the OH1 mail action, and I'm able to successfully send an email with an attachment.

### PR DESCRIPTION
**Pull request by J-N-K**
Fixes #6423 

Also adds filename and description to attachments.

Signed-off-by: Jan N. Klug <jan.n.klug@rub.de>

### CODE REVIEW

File: bundles/org.openhab.binding.mail/src/main/feature/feature.xml [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
     <feature name="openhab-binding-mail" description="Mail Binding" version="${project.version}">
         <feature>openhab-runtime-base</feature>
+        <feature>openhab-core-automation</feature>
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.mail/src/main/feature/feature.xml]: Imho this isn't the right fix.
The new rule engine must not be a prerequisite for using this binding.
Afaik, it currently only has an optional import to the annotations and nothing else.

I haven't dived into your analysis in the issue though, I have to admit.

Comment (J-N-K) [reply bundles/org.openhab.binding.mail/src/main/feature/feature.xml]: I don't understand it fully, too. 

But without this feature (or one of it's bundles) sending multi-part mails is not possible. Installing this feature made it work in two different setups.

Comment (kaikreuzer) [reply bundles/org.openhab.binding.mail/src/main/feature/feature.xml]: Maybe it is some dependency that the new rule engine brings with it, which is missing. A specific version of Apache Commons or whatever. I think we have to investigate it a bit further...

**Comment by kaikreuzer** [comment]
Superseded by https://github.com/openhab/openhab2-addons/pull/6494.
FYI, I have also included your other small changes.

**Pull request closed without merging**